unity-client/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarModifiers/AvatarModifierArea.cs

```diff
@@ -61,19 +61,58 @@ public override IEnumerator ApplyChanges(string newJson)
 
     private void OnDestroy()
     {
-        Destroy(collider);
+        RemoveAllModifiers();
+    }
+
+    private void Update()
+    {
+        if (model?.area == null)
+        {
+            return;
+        }
+
+        // Find avatars currently on the area
+        HashSet<GameObject> newAvatarsInArea = new HashSet<GameObject>(DetectAllAvatarsInArea());
+
+        // Call event for avatars that just entered the area
+        foreach (GameObject avatarThatEntered in newAvatarsInArea.Except(avatarsInArea))
+        {
+            OnAvatarEnter?.Invoke(avatarThatEntered);
+        }
+
+        // Call events for avatars that just exited the area
+        foreach (GameObject avatarThatExited in avatarsInArea.Except(newAvatarsInArea))
+        {
+            OnAvatarExit?.Invoke(avatarThatExited);
+        }
+
+        avatarsInArea = newAvatarsInArea;
     }
 
-    private void OnTriggerEnter(Collider other)
+    private GameObject[] DetectAllAvatarsInArea()
     {
-        Transform parent = other.transform.parent;
-        OnAvatarEnter?.Invoke(parent.gameObject);
+        if (entity?.gameObject == null)
+        {
+            return new GameObject[0];
```

Comment:
I think we could return null here instead of generating a new empty array, I just looked it up and it seems an empty array actually allocates memory in .NET, so to be on the safe side let's return null here.